rfc/kubernetes-ui-enhancements/kubernetes-ui-enhancements.md

```diff
@@ -0,0 +1,260 @@
+- Each UI tab has a dedicated button for directly deploying an unvalidated manifest. These button labels will be updated to reflect the tab name. A future iteration could offer a UI that scopes deployment objects to the tab’s given type.
+- Region and Availability Filters can be consolidated to a single filter for Namespaces.
+
+#### Technical Approach
+
+When creating an application, users will be presented with a checkbox for enabling “Provider Specific Terminology”. Upon enabling, users will select the terminology configuration to apply. This will allow each application team to determine what configuration works best for their mix of cloud providers. 
+
+By default, this will not be enabled for existing applications. This will ensure backwards compatibility with pre-existing applications and prevent unexpected terminology changes. 
+
+Initial rollout of this functionality would provide a terminology set tied to the Kubernetes provider but be generic enough to support extension to other cloud providers.
+
+Ideally, this approach will not degrade the UI experience for non-Kubernetes or mix provider setups. It will give users control on an application level and will improve the Kubernetes experience while remaining extensible.
+
+#### Concerns
+
+This approach may introduce a visual flicker during page load. Additionally, making UI terminology configurable at the application level is higher in complexity than an all or nothing approach such as a feature flag option.
+
+##### API
+
+Changes to terminology will be restricted to the UI. Where a mismatch would be introduced by updating a term in the UI, the API documentation will be updated to make note of the difference. Leaving endpoints unchanged will mitigate the risk of breaking current API consumers.
+
+##### Documentation
+
+Codelabs, API documentation, and user guides across Spinnaker.io will require updating. Terminology changes will need to be communicated to the community in advance through:
+
+- Terminology RFC
+- The Kubernetes SIG
+- Release Notes
+
+## Phase 2
+
+The scope of this project’s first phase has been detailed above. There exists a set of additional Kubernetes objects that may be helpful to expose to Spinnaker users. This section details a second phase of possible enhancements.
+
+## Custom CRDs
+
+Spinnaker currently offers [extension points](https://www.spinnaker.io/guides/developer/crd-extensions/) for advanced users seeking to accomplish a deeper integration with their own Custom Resource Definitions. This functionality offers users the ability to override the default model class for a CRD, implement a custom naming strategy, or offer a custom UI. To accomplish this, users must be comfortable working in Spinnaker’s code base and running a layered build. The complexity of this approach may act as a significant obstacle. Approaches such as general resource browser may present a more accessible alternative.
```

Comment:
The plugin effort does aim to make this style of extension easier and not dependent on a layered build — it may not completely solve the complexity problem tho.

FYI @robzienert @christopherthielen @claymccoy